core/src/main/java/org/apache/gravitino/catalog/CatalogManager.java

```diff
@@ -677,11 +678,42 @@ public boolean dropCatalog(NameIdentifier ident, boolean force)
     } catch (NoSuchMetalakeException | NoSuchCatalogException ignored) {
       return false;
 
-    } catch (IOException e) {
+    } catch (GravitinoRuntimeException e) {
+      throw e;
+
+    } catch (Exception e) {
       throw new RuntimeException(e);
     }
   }
 
+  private boolean haveAvailableSchemas(
```

Comment:
```suggestion
  private boolean hasSchemas(
```

Reply:
has <=> available

Reply:
I assume that the "default" schema is not an available schema in this method